gcp/src/gcpContext.ts

```diff
@@ -17,17 +17,25 @@ export class GcpContext implements CloudContext {
   public constructor(@inject(ComponentType.RuntimeArgs) args: any[]) {
     this.providerType = "gcp";
 
-    this.runtime = {
-      event: args[0],
-      context: args[1],
-      callback: args[2],
-    };
+    const isBackgroundFunction = args && args[1].eventId;
+    if (isBackgroundFunction) {
+      this.runtime = new GcpBackgroundFunctionRuntime();
+      this.runtime.event = args[0];
```

Comment:
code duplication in this if/else method. 
      this.runtime.event = args[0];
      this.runtime.context = args[1];

Reply:
Done f4b732fe39b43b9cf4003d35fbe084bad905ab5d